modules/mono/editor/GodotTools/GodotTools.Core/StringExtensions.cs

```diff
@@ -58,10 +58,19 @@ public static bool IsAbsolutePath(this string path)
                    path.StartsWith(_driveRoot, StringComparison.Ordinal);
         }
 
-        public static string ToSafeDirName(this string dirName, bool allowDirSeparator = false)
+        public static string ToSafeDirName(this string dirName, bool allowDirSeparator = false, bool assemblyName = false)
         {
             var invalidChars = new List<string> { ":", "*", "?", "\"", "<", ">", "|" };
 
+            if (assemblyName)
+            {
+                // Chars that are not valid in an AssemblyName
+                invalidChars.Add("'");
+                invalidChars.Add(";");
+                invalidChars.Add(",");
+                invalidChars.Add("=");
+            }
+
```

Comment:
There's at least one place in C++ that is using the safe dir function and as a result would be different from this one:

https://github.com/godotengine/godot/blob/3695fe5a573678c6491a6a33f19f9329d3256a48/modules/mono/mono_gd/gd_mono.cpp#L300

https://github.com/godotengine/godot/blob/3695fe5a573678c6491a6a33f19f9329d3256a48/core/config/project_settings.cpp#L71-L77

We should add these changes to C++ as well, as a new function. And to keep it simple, we should replace all the other safe dir calls with the new one, even the ones used for the data dir.